src/main/java/seedu/address/logic/commands/deletestudentfromclasscommands/DeleteStudentFromClassByEmailCommand.java

```diff
@@ -0,0 +1,80 @@
+package seedu.address.logic.commands.deletestudentfromclasscommands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.function.Predicate;
+
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.messages.PersonMessages;
+import seedu.address.logic.messages.TutorialClassMessages;
+import seedu.address.model.Model;
+import seedu.address.model.module.ModuleCode;
+import seedu.address.model.module.ModuleTutorialPair;
+import seedu.address.model.module.TutorialClass;
+import seedu.address.model.person.Email;
+import seedu.address.model.person.Person;
+
+/**
+ * Deletes a student from a specified tutorial class, by identifying
+ * the student via their email.
+ */
+public class DeleteStudentFromClassByEmailCommand extends DeleteStudentFromClassCommand {
+    private final Predicate<Person> predicate;
+
+    private final Email email;
+
+    /**
+     * Adds a student to a class by email.
```

Comment:
I think you probably meant delete instead of add here. I think this is also seen in the other classes